### DISCUSSION TITLE
Version Control for Beatmaps

### DISCUSSION THREAD

**Discussion by 19peytonsmith** [ppy/osu#23707] [Ideas]
I think it would be neat to see a github-esque version control for beatmap contents for maps that have been updated more than once. Not sure on the exact way of implementing it, was thinking when you view a beatmap listing it will show basic changes lik3 changes with object count or ar/hp/cs/od but also going a little more advanced with what sections of the map was changed. Since this would couple nicely with showing the beatmap editor itself, maybe this portion of the feature would work best in the client rather than having a web beatmap displayer.

**Comment by peppy** [comment]
There has to already be an issue/discussion tracking this.

**Reply by 19peytonsmith** [reply]
I've tried searching for preexisting issues but I couldn't find one, perhaps you could link it?

**Reply by soopax** [reply]
https://github.com/ppy/osu/discussions/15372#discussioncomment-1561633

**Reply by bdach** [reply]
Also https://github.com/ppy/osu-web/issues/7302
